pkg/azurefile/controllerserver.go

```diff
@@ -132,11 +137,20 @@ func (d *Driver) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequest)
 
 	accountKey, err := GetStorageAccesskey(d.cloud, accountName, resourceGroupName)
 	if err != nil {
-		return nil, fmt.Errorf("no key for storage account(%s) under resource group(%s), err %v", accountName, resourceGroupName, err)
+		// Same case as above
+		// reference: https://github.com/container-storage-interface/spec/blob/master/spec.md#deletevolume
+		return &csi.DeleteVolumeResponse{}, nil
 	}
 
 	klog.V(2).Infof("deleting azure file(%s) rg(%s) account(%s) volumeID(%s)", fileShareName, resourceGroupName, accountName, volumeID)
 	if err := d.cloud.DeleteFileShare(accountName, accountKey, fileShareName); err != nil {
+		if e, ok := err.(check.AzureStorageServiceError); ok && e.StatusCode == http.StatusNotFound {
+			// Same case as above
+			// reference: https://github.com/container-storage-interface/spec/blob/master/spec.md#deletevolume
+			return &csi.DeleteVolumeResponse{}, nil
```

Comment:
if delete volume failed, it should not `return &csi.DeleteVolumeResponse{}, nil`, right?